### PR TITLE
docs: enhance API docs with information about results of code generation with generateFromString

### LINKED ISSUE #507: Enhance API docs with information about results of code generation with generateFromString using entrypoint
#### Reason/Context
The problem is described here https://github.com/asyncapi/playground/pull/79. It must be clear that returned data is not content but an object with content and metadata + what is this metadata, and that it is only with React renderer.

#### Description
Probably JSDoc must be only updated. Of course, some additional section in the readme would be also welcomed 😄

### PR DESCRIPTION
**Pull request by chinma-yyy**
**Description**

- Described a type of `TemplateRenderMetadata`
- Described a type of `TemplateRenderResult` 

**Related issue(s)**

Resolves #507

### CODE REVIEW

**Review by derberg — APPROVED** [review]

**Comment by derberg** [comment]
changed to `docs` as we do not have to trigger release

**Comment by derberg** [comment]
@Florence-Njeri @pratik2315 folks we need your approval to merge

**Comment by Florence-Njeri** [comment]
@derberg I'll review this before EOD

File: docs/api.md [hunk 1]
```diff
@@ -4,6 +4,22 @@ weight: 75
...
+
+<dl>
+<dt><a href="#Generator">Generator</a></dt>
+<dd></dd>
+</dl>
+
+## Typedefs
+
+<dl>
+<dt><a href="#TemplateRenderMetadata">TemplateRenderMetadata</a> : <code>Object</code></dt>
```
Comment (Florence-Njeri) [inline docs/api.md:17]: @derberg @chinma-yyy what do the #TemplateRenderMetadata and #TemplateRenderResults link to. They don't navigate to any section of this page.

**Review by Florence-Njeri — CHANGES_REQUESTED** [review]

File: docs/api.md (hunk 1 above)
Comment (Florence-Njeri) [reply docs/api.md:17]: This is what it will look like in the website:

Comment (chinma-yyy) [reply docs/api.md:17]: Actually it should render the object properties 🤔. Will look into it

**Comment by Florence-Njeri** [comment]
@chinma-yyy Do you have any updates regarding this PR?

**Comment by derberg** [comment]
@chinma-yyy please respons, otherwise we will have to close this PR as it is blocking others to work on it

**Pull request closed without merging**